### ISSUE TITLE
Services not being blocked

### ISSUE THREAD

**Issue by thunderclap82** [AdguardTeam/AdGuardHome#4917]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not ask a question

### Operating system type

Linux, Other (please mention the version in the description)

### CPU architecture

64-bit ARM

### Installation

Other (please mention in the description)

### Setup

On one machine

### AdGuard Home version

v0.108.0-a.278+88812f05

### Description

Running the latest edge build on Raspberry Pi (snap) to fix bug [this bug](https://github.com/AdguardTeam/AdGuardHome/issues/4904), however blocking of services (such as Youtube) no longer works. Access to service is still accessible on devices set to be blocked.

**Comment by ainar-g** [comment]
Hello.  Please follow the [steps in the FAQ][1] to diagnose the issue.  In particular, make sure that you can actually see the queries for these resources (YouTube, etc.) in the Query log and that clients are correctly identified there.

[1]: https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#doesntblock

**Comment by thunderclap82** [comment]
I’m not sure what I did but I just start testing different settings and it suddenly started working again. So not a big but user error somewhere.

**Comment by theschles** [comment]
@ainar-g  please re-open this?

### Prerequisites

I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

I have searched other issues and found no duplicates

I want to report a bug and not ask a question

### Operating system type

Docker on RaspbianOS

### CPU architecture

`uname -a` :
Linux 6.1.21-v7+ #1642 SMP Mon Apr  3 17:20:52 BST 2023 armv7l GNU/Linux

### Setup

On one machine

### AdGuard Home version

v0.107.38

### Description

tl;dr:
* In AGH I enabled Client filter for a particular IP with the YouTube service blocked
* I then try surfing to Youtube.com on that particular IP -- and the Youtube home page loads
* Refresh again and again and again -- and the Youtube home page STILL loads
* I look at AGH's Query Log -- and the Query Log shows repeatedly blocking YouTube for that particular IP
* HELP!?!


Longer description:
I have two sons -- each of whom have a school-issued Chromebook.  They're watching YouTube whenever we're not looking...which putting their grades in danger.

I installed AGH on Docker on one of my Raspberry Pi's.

I set up a Client entry in AGH for the IP address of one of the boys' Chromebooks -- setting YouTube to be a disabled service.

I now open up Youtube in Chrome on one son's Chromebook -- and the Youtube page loads just fine.  It continues to load just fine as a refresh and refresh and refresh.

I tried disabling the Youtube block-service button in the Client setting, saving, then re-enabling the Youtube block-service button and saving.  Still not blocked.

I now look at AGH's Query Log and -- sure enough -- AGH is officially blocking the name-to-IP resolution.  Somehow still the Chromebook can do the name-to-IP resolution.

I then wondered if the problem was DNS TTL -- so I set AGH's DNS Maximum TTL to 300 (seconds).

No change: Youtube still opens just fine on my son's Chromebook.

HELP?!?!?

**Comment by theschles** [comment]
Never mind, I think I know what I did wrong: I configured DNS setting for `NXDOMAIN` -- and I think that caused it to fail over to another DNS server.

**Comment by 2507rkt3** [comment]
I'm experiencing the same thing. 

@theschles - what settings did you use to get it to work?
